test/common/config/grpc_mux_impl_test.cc

```diff
@@ -33,16 +34,17 @@ namespace {
 // is provided in [grpc_]subscription_impl_test.cc.
 class GrpcMuxImplTest : public testing::Test {
 public:
-  GrpcMuxImplTest()
-      : async_client_(new Grpc::MockAsyncClient()), timer_(new Event::MockTimer()), time_source_{} {
+  GrpcMuxImplTest() : async_client_(new Grpc::MockAsyncClient()), time_source_{} {}
+
+  void setup() {
     EXPECT_CALL(dispatcher_, createTimer_(_)).WillOnce(Invoke([this](Event::TimerCb timer_cb) {
       timer_cb_ = timer_cb;
+      timer_ = new Event::MockTimer();
```

Comment:
Since the newly introduced tests do not call `setup()`, and `timer_` was previously initialized as `new Event::MockTimer()` at constructor, this makes sure no leak. Caught by LeakSanitizer in this build: https://circleci.com/gh/envoyproxy/envoy/84216.